sigstore/verify/verifier.py

```diff
@@ -108,6 +119,156 @@ def _from_trust_config(cls, trust_config: ClientTrustConfig) -> Verifier:
             trusted_root=trust_config.trusted_root,
         )
 
+    def _verify_signed_timestamp(
+        self, timestamp_response: TimeStampResponse, signature: bytes
+    ) -> Union[None, TimestampVerificationResult]:
+        """
+        Verify a Signed Timestamp using the TSA provided by the Trusted Root.
+        """
+        cert_authorities = self._trusted_root.get_timestamp_authorities()
+        for certificate_authority in cert_authorities:
+            certificates = certificate_authority.certificates(allow_expired=True)
+
+            builder = VerifierBuilder()
+            for certificate in certificates[:-1]:
+                builder.add_intermediate_certificate(certificate)
+            builder.add_root_certificate(certificates[-1])
```

Comment:
This is a minor thing, but technically we can do `add_root_certificate` for all of these, since these are all in the "root of trust" regardless of whether they're intermediates or not. That'll result in a slightly shorter chain when verifying.